### PR TITLE
[SPARK-23119][SS] Minor fixes to V2 streaming APIs

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

- Added `@InterfaceStability.Evolving` annotations
- Improved docs.

## How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by tdas** [comment]
@jose-torres @zsxwing Can you take a look.

**Comment by jose-torres** [comment]
LGTM

**Comment by tdas** [comment]
jenkins retest this please.

File: sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/MicroBatchReader.java [hunk 1]
```diff
@@ -25,7 +26,11 @@
 /**
  * A mix-in interface for {@link DataSourceV2Reader}. Data source readers can implement this
  * interface to indicate they allow micro-batch streaming reads.
+ *
+ * Note: This class currently extends {@link BaseStreamingSource} to maintain compatibility with
+ * DataSource V1 APIs. This will be extension will be removed once we get rid of V1 completely.
```
Comment (zsxwing) [inline sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/MicroBatchReader.java]: ditto

File: sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/ContinuousReader.java [hunk 2]
```diff
@@ -27,11 +28,15 @@
  * interface to allow reading in a continuous processing mode stream.
  *
  * Implementations must ensure each read task output is a {@link ContinuousDataReader}.
+ *
+ * Note: This class currently extends {@link BaseStreamingSource} to maintain compatibility with
+ * DataSource V1 APIs. This will be extension will be removed once we get rid of V1 completely.
```
Comment (zsxwing) [inline sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/ContinuousReader.java]: nit: This ~~will be~~ extension

File: sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/Offset.java [hunk 3]
```diff
@@ -17,12 +17,20 @@
...
- * restart checkpoints. Sources should provide an Offset implementation which they can use to
- * reconstruct the stream position where the offset was taken.
+ * An abstract representation of progress through a {@link MicroBatchReader} or
+ * {@link ContinuousReader}.
+ * During execution, offsets provided by the data source implementation will be logged and used as
+ * restart checkpoints. Each source should provide an offset implementation which the source can use
+ * to reconstruct a position in the stream up to which data has been seen/processed.
+ *
+ * Note: This class currently extends {@link org.apache.spark.sql.execution.streaming.Offset} to
+ * maintain compatibility with DataSource V1 APIs. This will be extension will be removed once we
```
Comment (zsxwing) [inline sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/reader/Offset.java]: ditto

**Review by zsxwing — COMMENTED** [review]
LGTM except some nits

**Comment by smurakozi** [comment]
I think this change is OK, except the nits zsxwing already noted.

**Review by smurakozi — COMMENTED** [review]
I think this change is OK, except the nits zsxwing already noted.

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.3.

**Pull request closed without merging**